core/src/main/java/org/apache/metamodel/data/AbstractRowBuilder.java

```diff
@@ -68,7 +71,7 @@ public AbstractRowBuilder(Column[] columns) {
 
     @Override
     public final Row toRow() {
-        return new DefaultRow(new SimpleDataSetHeader(_columns), _values);
+        return new DefaultRow(new SimpleDataSetHeader(Arrays.stream(_columns).map(SelectItem::new).collect(Collectors.toList())), _values);
```

Comment:
Array and List have a slight different semantic.
My reasoning was, if i can grasp the code within a minute, i changed internal methods, otherwise preserve it.

This is especially the case with the XML-Dom-related stuff.